LandSandBoat/server
# [core] [SMN] [lua] Implement "Petskill" system, add basic Siren implementation

**_I affirm:_**

## What does this pull request do?

adds "Petskill" system, which is largely a copy paste of mobskill, but distinctly different in how categories are used for action packets. This allows for proper usage of pet models & pet mobskill IDs in packets.
Implements Siren at a basic level -- welt, favor only.

## Steps to test these changes

- Summon siren
- Use Avatar's Favor
- See `!getmod subtle_blow` climbs with time
- Use Welt on a mob and have it work + deal damage

## Notes
petskill interface is subject to change as I am not sure what is needed or not. Ideally the interface looks identical to any ability in lua but core can be more lightweight.
